cmd/profilecli/main.go

```diff
@@ -82,6 +82,8 @@ func main() {
 	queryBlocksCmd := app.Command("query-blocks", "Query on local/remote blocks")
 	queryBlocksSeriesCmd := queryBlocksCmd.Command("series", "Request series labels on local/remote blocks")
 	queryBlocksSeriesParams := addQueryBlocksSeriesParams(queryBlocksSeriesCmd)
+	queryBlocksMergeCmd := queryBlocksCmd.Command("merge", "Request merged profile.")
+	queryBlocksMergeParams := addQueryBlocksMergeParams(queryBlocksMergeCmd)
```

Comment:
Let's design the CLI interface first. I believe that `merge` might be a touch confusing name (even If we already use it in `query` subcommand).

I propose the following interface:

```
profilecli query merge.  // Already exists. Should be hidden and replaced in docs with "profile".
profilecli query profile // Alias for "merge".
profilecli query series. // Existing subcommand.
profilecli query go-pgo  // Queries pprof for Go PGO.
etc.
```

Now, in the command handler, we check whether `--block` flag is specified. There's a common practice to use singular form for flags that accept multiple values; the flag should be specified multiple times:
```bash
profilecli query series --block=A

profilecli query series \
  --block=A \
  --block=B \
```

Next, let's make `query` subcommand to support storage backend configuration ([this is very easy](https://github.com/grafana/pyroscope/pull/3610#discussion_r1791194740)).

Finally, let's remove `query-blocks` subcommand.

Reply:
So after inspecting this, you may see a draft for both solutions:
* https://github.com/grafana/pyroscope/pull/3623
* https://github.com/grafana/pyroscope/pull/3625

I agree adding to `profilecli query` may make more sense. But on the other hand, I think it makes doc more difficult to express.

Here we have the `--help`. Marking with `>` flags for pyroscope server mode, with `<` those flags for single blocks (require the `--block`), and finally marking with `=` those flags that are used for both modes:
```
Request merged profile.

Flags:
  -h, --help                     Show context-sensitive help (also try --help-long and --help-man).
      --version                  Show application version.
  -v, --verbose                  Enable verbose logging.
>      --url="http://localhost:4040"  
                                 URL of the profile store.
>      --tenant-id=""             The tenant ID to be used for the X-Scope-OrgID header.
>      --username=""              The username to be used for basic auth.
>      --password=""              The password to be used for basic auth.
>      --protocol=connect         The protocol to be used for communicating with the server.
>      --from="now-1h"            Beginning of the query.
>      --to="now"                 End of the query.
=      --query="{}"               Label selector to query.
<      --local-path="./data/anonymous/local"  
                                 Path to blocks directory.
<      --bucket-name=BUCKET-NAME  The name of the object storage bucket.
<      --object-store-type="gcs"  The type of the object storage (e.g., gcs).
<      --block=BLOCK ...          Block ids to query on (accepts multiples)
<      --block-tenant-id=BLOCK-TENANT-ID  
                                 Tenant id of the queried block for remote bucket
=      --output="console"         How to output the result, examples: console, raw, pprof=./my.pprof
=      --profile-type="process_cpu:cpu:nanoseconds:cpu:nanoseconds"  
                                 Profile type to query.
=      --stacktrace-selector=STACKTRACE-SELECTOR ...  
                                 Only query locations with those symbols. Provide multiple times starting with the root
```
So at the end you have a set of shared flags, some that will need some values if you use `--block`, and others that are only needed in the absence of that flag.

So that makes me lean more towards `admin blocks query` solution, which in terms of docs feels more precise:
```
Request merged profile on local/remote block.

Flags:
  -h, --help                     Show context-sensitive help (also try --help-long and --help-man).
      --version                  Show application version.
  -v, --verbose                  Enable verbose logging.
      --path="./data/anonymous/local"  
                                 Path to blocks directory
      --bucket-name=BUCKET-NAME  The name of the object storage bucket.
      --object-store-type="gcs"  The type of the object storage (e.g., gcs).
      --block=BLOCK ...          Block ids to query on (accepts multiples)
      --tenant-id=TENANT-ID      Tenant id of the queried block for remote bucket
      --query="{}"               Label selector to query.
      --output="console"         How to output the result, examples: console, raw, pprof=./my.pprof
      --profile-type="process_cpu:cpu:nanoseconds:cpu:nanoseconds"  
                                 Profile type to query.
      --stacktrace-selector=STACKTRACE-SELECTOR ...  
                                 Only query locations with those symbols. Provide multiple times starting with the root
```